Sources/Layout/MessagesCollectionViewFlowLayout.swift

```diff
@@ -45,31 +44,51 @@ open class MessagesCollectionViewFlowLayout: UICollectionViewFlowLayout {
     open var messageLabelFont: UIFont {
         didSet {
             emojiLabelFont = messageLabelFont.withSize(2 * messageLabelFont.pointSize)
+            textMessageAttributesCache.removeAllObjects()
+            attributedTextMessageAttributesCache.removeAllObjects()
+            emojiMessageAttributesCache.removeAllObjects()
         }
     }
 
-    /// The font to be used by `TextMessageCell` for `MessageData.emoji(String)` case.
-    ///
-    /// Note: The default value of this property is 2x the `messageLabelFont`.
     internal var emojiLabelFont: UIFont
 
-    /// Determines the maximum number of `MessageCollectionViewCell` attributes to cache.
-    ///
-    /// Note: The default value of this property is 500.
-    open var attributesCacheMaxSize: Int = 500 {
-        didSet {
-            layoutContextCache.countLimit = attributesCacheMaxSize
-        }
+    // MARK: - Layout
+
+    open var textMessageLayout = TextMessageLayout() {
+        didSet { textMessageAttributesCache.removeAllObjects() }
+    }
+
+    open var attributedTextMessageLayout = TextMessageLayout() {
+        didSet { attributedTextMessageAttributesCache.removeAllObjects() }
+    }
+
+    open var emojiMessageLayout = TextMessageLayout() {
+        didSet { emojiMessageAttributesCache.removeAllObjects() }
+    }
+
+    open var photoMessageLayout = MediaMessageLayout() {
+        didSet { photoMessageAttributesCache.removeAllObjects() }
     }
 
-    typealias MessageID = NSString
-    
-    /// The cache for `MessageCellLayoutContext`.
-    /// The key is the `messageId` of the `MessageType`.
-    fileprivate var layoutContextCache = NSCache<MessageID, MessageCellLayoutContext>()
+    open var videoMessageLayout = MediaMessageLayout() {
+        didSet { videoMessageAttributesCache.removeAllObjects() }
+    }
+
+    open var locationMessageLayout = MediaMessageLayout() {
+        didSet { locationMessageAttributesCache.removeAllObjects() }
+    }
 
-    /// The `MessageCellLayoutContext` for the current cell.
-    internal var currentLayoutContext: MessageCellLayoutContext!
+    open var customMessageLayout: MessageCellLayout? {
```

Comment:
We only support one kind of custom cell?

Reply:
I agree. I need to think about this